contracts/Aavegotchi/facets/ERC1155BuyOrderFacet.sol

```diff
@@ -0,0 +1,264 @@
+            require(erc1155BuyOrder.timeCreated + erc1155BuyOrder.duration >= block.timestamp, "ERC1155BuyOrder: Already expired");
+        }
+        require(erc1155BuyOrder.quantity >= _quantity, "ERC1155BuyOrder: Sell amount should not be larger than quantity of the buy order");
+
+        uint256 cost = _quantity * erc1155BuyOrder.priceInWei;
+        require(cost >= 1e15, "ERC1155BuyOrder: execution cost should be 0.001 GHST or larger");
+
+        IERC1155 erc1155Token = IERC1155(erc1155BuyOrder.erc1155TokenAddress);
+        require(erc1155Token.balanceOf(sender, erc1155BuyOrder.erc1155TokenId) >= _quantity, "ERC1155Marketplace: Not enough ERC1155 token");
+
+        erc1155BuyOrder.quantity -= _quantity;
+        erc1155BuyOrder.lastTimePurchased = block.timestamp;
+
+        BaazaarSplit memory split = LibSharedMarketplace.getBaazaarSplit(cost, new uint256[](0), [10000, 0]);
+        LibSharedMarketplace.transferSales(
+            SplitAddresses({
+                ghstContract: s.ghstContract,
+                buyer: address(this),
+                seller: sender,
+                affiliate: address(0),
+                royalties: new address[](0),
+                daoTreasury: s.daoTreasury,
+                pixelCraft: s.pixelCraft,
+                rarityFarming: s.rarityFarming
+            }),
+            split
+        );
+
+        // new sub buy order
+        s.nextERC1155BuyOrderId++;
+        uint256 subBuyOrderId = s.nextERC1155BuyOrderId;
+        s.erc1155BuyOrders[subBuyOrderId] = ERC1155BuyOrder({
+            buyOrderId: subBuyOrderId,
+            buyer: erc1155BuyOrder.buyer,
+            erc1155TokenAddress: erc1155BuyOrder.erc1155TokenAddress,
+            erc1155TokenId: erc1155BuyOrder.erc1155TokenId,
+            priceInWei: erc1155BuyOrder.priceInWei,
+            quantity: _quantity,
+            sourceBuyOrderId: _buyOrderId,
+            timeCreated: block.timestamp,
+            lastTimePurchased: block.timestamp,
+            duration: erc1155BuyOrder.duration,
+            completed: true,
+            cancelled: false
+        });
+        if (erc1155BuyOrder.quantity == 0) {
+            erc1155BuyOrder.completed = true;
+            LibBuyOrder.removeERC1155BuyOrder(_buyOrderId);
+        }
+
+        // ERC1155 transfer
+        if (erc1155BuyOrder.erc1155TokenAddress == address(this)) {
+            LibItems.removeFromOwner(sender, erc1155BuyOrder.erc1155TokenId, _quantity);
+            LibItems.addToOwner(erc1155BuyOrder.buyer, erc1155BuyOrder.erc1155TokenId, _quantity);
+            IEventHandlerFacet(s.wearableDiamond).emitTransferSingleEvent(
+                address(this),
+                sender,
+                erc1155BuyOrder.buyer,
+                erc1155BuyOrder.erc1155TokenId,
+                _quantity
+            );
+            LibERC1155.onERC1155Received(address(this), sender, erc1155BuyOrder.buyer, erc1155BuyOrder.erc1155TokenId, _quantity, "");
```

Comment:
I think you also need to handle the baazaar hook to delist the item from the baazaar, if it was listed @orionstardust

Reply:
Updated.